tools/dts.js

```diff
@@ -205,44 +209,48 @@ function parseExportFrom(content, currentFileName, queue, baseDir, projDir) {
     return content.replace(regExportFrom, '');
 }
 
+function parseImportFrom(content, currentFileName, queue, baseDir, projDir) {
+    var matches;
+    while ((matches = regImportFrom.exec(content))) {
+        var fromFileName = parseFrom(matches[1].trim(), currentFileName, baseDir, projDir);
+        enqueue(queue, fromFileName);
+    }
+    return content.replace(regImportFrom, '');
+}
+
 function process(baseDir, queue, index, projDir) {
     var item = queue[index];
     var currentFileName = item.filename;
     var file = fs.readFileSync(currentFileName);
     var content = file.toString();
 
-    // 1. Remove single line comments
-    content = content.replace(singleLineComment, '');
-
-    // 2. Process 'export ... from ...;'
+    // 1. Process 'export ... from ...;'
     content = parseExportFrom(content, currentFileName, queue, baseDir, projDir);
 
-    // 3. Remove imports
-    content = content.replace(regImportFrom, '');
-
-    // 4. Parse classes and interfaces
-    content = parseClasses(content, item.elements);
-
-    // 5. Parse functions
-    content = parseFunctions(content, item.elements);
-
-    // 6. Parse enum
-    content = parseEnum(content, item.elements);
-
-    // 7. Parse type
-    content = parseType(content, item.elements);
+    // 2. Remove imports
+    content = parseImportFrom(content, currentFileName, queue, baseDir, projDir);
 
-    // 8. Parse const
-    content = parseConst(content, item.elements);
+    // 3. Parse all the public elements
+    content = [parseClasses, parseFunctions, parseEnum, parseType, parseConst, parseExport].reduce(
+        (c, func) => func(c, item.elements),
+        content
+    );
 
-    // 9. Parse Export
-    content = parseExport(content, item.elements);
+    // 4. Remove single line comments
+    content = content.replace(singleLineComment, '');
 
     if (content.trim() != '') {
         err('File ' + currentFileName + ' contains unrecognized content:\r\n' + content);
     }
 }
 
+function publicElement(element) {
+    return element.map(x => {
+        x.published = true;
```

Comment:
is this really necessary if you are returning x.text here?

Reply:
oh nevermind I see that you are using the published prop in another parts of the code